typed-racket-more/typed/net/url.rkt

```diff
@@ -1,86 +1,70 @@
 #lang typed/racket/base
 
-(require typed/private/utils)
-
-(require-typed-struct/provide path/param ([path : (U String 'up 'same)] [param : (Listof String)]) net/url)
-
-(require-typed-struct/provide
- url ([scheme : (Option String)]
-      [user : (Option String)]
-      [host : (Option String)]
-      [port : (Option Integer)]
-      [path-absolute? : Boolean]
-      [path : (Listof path/param)]
-      [query : (Listof (Pair Symbol (Option String)))]
-      [fragment : (Option String)])
- net/url)
-
-(require/opaque-type URL-Exception url-exception? net/url)
-(provide URL-Exception url-exception?)
-
-(require/opaque-type HTTP-Connection http-connection? net/url)
-(provide HTTP-Connection http-connection?)
-
-(define-type-alias PortT (case-lambda (url -> Input-Port) (url (Listof String)-> Input-Port)))
```

Comment:
This is why:

```racket
Welcome to Racket v6.1.1.
> (require typed/net/url)
> (:print-type get-impure-port)
PortT
>
```

As long as type alias printing works the way it does, providing type aliases needs to be done with a lot of caution. They effectively make the types opaque. As far as I know, there's no way to print the "underlying" type the alias refers to.